### ISSUE TITLE
Hyper-parameters are inconsistent with Table 8

### ISSUE THREAD

**Issue by ZeguanXiao** [sanyalsunny111/LLM-Inheritune#4]
Hyper-parameters in the [training script](https://github.com/sanyalsunny111/LLM-Inheritune/blob/26fff62c50f5809ac12a523ba4fc7e9f6e6444b5/lit-gpt/Training/train.py#L45) are inconsistent with paper, including batch size, lr scheduler and weight decay. The train epoch is also missing, instead of a max_iters. In my experience, it is hard to train a 1.5B model on 1B tokens in 8 hours. Is there something wrong in code or paper?

<img width="445" alt="截屏2024-06-24 14 05 08" src="https://github.com/sanyalsunny111/LLM-Inheritune/assets/38279341/406fff2f-7a75-4bc7-bc49-0b884c5e6bdb">

**Comment by sanyalsunny111** [comment]
Yes, that's just a sample script provided for running an experiment for full reproducibly please use the paper's hyperparameters.

**Comment by ZeguanXiao** [comment]
My issue is that the training time is longer than paper said. When I run the code in a L40 48G GPU, it takes 37.8 seconds to run 100 iterations, which corresponds to 100 * 4096 = 409,600 tokens. So I think the 1B * 8 epoches = 8B tokens will use much more times to run than 8 hours. Do I misunderstanding anything?

**Comment by sanyalsunny111** [comment]
The training time is a function of the GPUs (also other variables) and I see you are using L40 whereas we have used A6000 I guess this could be the reason. But I would still like to know your GPU hours?

**Comment by ZeguanXiao** [comment]
The estimated cost is 8,000,000,000 tokens / 409,600 * 37.8 seconds / 3600 ≈ 205 GPU hours.
Below is a screenshot that shows the time of running.

<img width="575" alt="截屏2024-06-26 12 22 16" src="https://github.com/sanyalsunny111/LLM-Inheritune/assets/38279341/dfdc1d94-2815-48eb-85f1-16d25886581e">

**Comment by ZeguanXiao** [comment]
Hi, @sanyalsunny111, I would appreciate it if you could rerun the example code and time it. I'm surprised at the large difference in GPU hours between L40 and A6000, and I wonder if I'm doing something wrong.

**Comment by sanyalsunny111** [comment]
Hi @ZeguanXiao Sure, I am gonna do that. But I am currently on leave from my university for summer internship, so, I don't have access to my lab resources (A6000 which I used). Once I go back I will rerun this again.
